plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.js

```diff
@@ -16,17 +17,36 @@ export default class ChatChannelUnreadIndicator extends Component {
   }
 
   get unreadCount() {
+    if (this.#onlyMentions() && this.#hasChannelMentions()) {
+      return this.args.channel.tracking.mentionCount;
+    }
     return this.args.channel.tracking.unreadCount;
   }
 
   get isUrgent() {
+    if (this.#onlyMentions()) {
+      return this.#hasChannelMentions();
+    }
     return (
-      this.args.channel.isDirectMessageChannel ||
-      this.args.channel.tracking.mentionCount > 0
+      this.args.channel.isDirectMessageChannel || this.#hasChannelMentions()
     );
   }
 
   get showUnreadCount() {
+    if (this.#onlyMentions()) {
+      return this.#hasChannelMentions();
+    }
     return this.args.channel.isDirectMessageChannel || this.isUrgent;
   }
+
+  #hasChannelMentions() {
+    return this.args.channel.tracking.mentionCount > 0;
+  }
+
+  #onlyMentions() {
+    return (
+      this.currentUser.user_option.chat_header_indicator_preference ===
+      "only_mentions"
```

Comment:
Maybe import the const of user preferences, or better create a constants file in lib/ and we can start putting stuff in it. Up to you.